Tests for chat.cs

daanl opened the pull request.

daanl committed:
Adding tests for chat.cs

davidfowl commented on JabbR.Test/ChatFacts.cs:
```diff
@@ -0,0 +1,143 @@
+                this.MockedResourceProcessor = mockedResourceProcessor;
+                this.MockedChatService = mockedChatService;
+                this.Repository = repository;
+                this.MockedConnection = connection;
```
Remove "this." and make those properties.

daanl replied:
Done

davidfowl commented on JabbR/Models/InMemoryRepository.cs:
```diff
@@ -71,17 +69,17 @@ public void Dispose()
         public ChatUser GetUserById(string userId)
         {
-            return _users.FirstOrDefault(u => u.Id.Equals(userId, StringComparison.OrdinalIgnoreCase));
+            return _users.Where(x => x.Id != null).FirstOrDefault(u => u.Id.Equals(userId, StringComparison.OrdinalIgnoreCase));
```
Why not just an and?

`FirstOrDefault(u => u.Id != null && u.Id.Equals(userId .....))`

daanl replied:
Fixed now

davidfowl commented on JabbR/packages.config:
```diff
@@ -16,7 +16,7 @@
   <package id="json2" version="1.0" />
   <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" />
-  <package id="Newtonsoft.Json" version="4.0.4" />
+  <package id="Newtonsoft.Json" version="4.0.5" />
```
How come?

daanl replied:
Installed Newtonsoft by nuget for Jabbr.Tests, the compile broke cuz it was 4.0.4 so i updated it by nuget. for Jabbr

daanl committed:
Changes simplify != null in InMemoryRepository.cs
Making properties instead of fields and ensuring these cannot be set externally
